flake.nix

```diff
@@ -54,9 +50,9 @@
           type = "app";
           program = self'.packages.default;
         };
-        packages.rio = pkgs.callPackage mkRio {rust-toolchain = rust-toolchain;};
+        packages.rio = pkgs.callPackage mkRio {rust-toolchain = pkgs.rust-bin.stable.latest.minimal;};
 
-        devShells.msrv = mkDevShell rust-toolchain;
+        devShells.msrv = mkDevShell (pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml);
```

Comment:
Hm... we can do this but then I'd like to add the `rust-analyzer` component to the toolchain (`rust-toolchain.toml`). What do you say?

Reply:
> I disagree. If rust-toolchain.toml represents the minimal set of Rust software required to build the project, rust-analyzer is not required.

True. Sounds reasonsble. According to this SO answer (https://stackoverflow.com/a/62419829) it does look like as if the main purpose of the rust toolchain is to install the _required_ tools to build the rust project and that you can optionally add other compononents.

So yeah, adding `rust-analyzer` and `rustfmt` just to the devshell if fine for me.

I'd like to try your PR later out and then my approvement if everything works 👍